yt/frontends/sph/data_structures.py

```diff
@@ -104,19 +129,25 @@ def _initialize_index(self):
 
     def _generate_kdtree(self, fname):
         from cykdtree import PyKDTree
-        if os.path.exists(fname):
-            mylog.info('Loading KDTree from %s' % os.path.basename(fname))
-            kdtree = PyKDTree.from_file(fname)
-            if kdtree.data_version != self.ds._file_hash:
-                mylog.info('Detected hash mismatch, regenerating KDTree')
-            else:
-                self._kdtree = kdtree
-                return
+        if fname is not None:
+            if os.path.exists(fname):
+                mylog.info('Loading KDTree from %s' % os.path.basename(fname))
+                kdtree = PyKDTree.from_file(fname)
+                if kdtree.data_version != self.ds._file_hash:
+                    mylog.info('Detected hash mismatch, regenerating KDTree')
+                else:
+                    self._kdtree = kdtree
+                    return
```

Comment:
Modified this so it does not attempt to save the test dataset kdtree's to a file

Reply:
good catch!